custom_components/tahoma/climate_devices/atlantic_heat_recovery_ventilation.py

```diff
@@ -190,50 +180,78 @@ def preset_modes(self) -> Optional[List[str]]:
     async def async_set_preset_mode(self, preset_mode: str) -> None:
         """Set the preset mode of the fan."""
 
-        if preset_mode == PRESET_COMFORT:
+        if preset_mode == PRESET_AUTO:
             await self.async_execute_command(
                 COMMAND_SET_VENTILATION_CONFIGURATION_MODE, "comfort"
             )
+            VENTILATION_MODE_STATE["prog"] = "off"
             await self.async_execute_command(
-                COMMAND_SET_VENTILATION_MODE, {"prog": "off"}
+                COMMAND_SET_VENTILATION_MODE, VENTILATION_MODE_STATE
             )
 
         if preset_mode == PRESET_PROG:
             await self.async_execute_command(
                 COMMAND_SET_VENTILATION_CONFIGURATION_MODE, "standard"
             )
+            VENTILATION_MODE_STATE["prog"] = "on"
             await self.async_execute_command(
-                COMMAND_SET_VENTILATION_MODE, {"prog": "on"}
+                COMMAND_SET_VENTILATION_MODE, VENTILATION_MODE_STATE
             )
 
         if preset_mode == PRESET_MANUAL:
             await self.async_execute_command(
                 COMMAND_SET_VENTILATION_CONFIGURATION_MODE, "standard"
             )
+            VENTILATION_MODE_STATE["prog"] = "off"
             await self.async_execute_command(
-                COMMAND_SET_VENTILATION_MODE, {"prog": "off"}
-            )
-
-        if preset_mode == PRESET_ECO:
-            await self.async_execute_command(
-                COMMAND_SET_VENTILATION_CONFIGURATION_MODE, "eco"
-            )
-            await self.async_execute_command(
-                COMMAND_SET_VENTILATION_MODE, {"prog": "off"}
+                COMMAND_SET_VENTILATION_MODE, VENTILATION_MODE_STATE
             )
+        time.sleep(0.5)
```

Comment:
This is not allowed I believe, in async context. However I could have a look at this in my PR eventually.